core/commands/pin/remotepin.go

```diff
@@ -173,6 +175,34 @@ NOTE: a comma-separated notation is supported in CLI for convenience:
 		if err != nil {
 			return err
 		}
+		//ifconfig.co
+		checkConnection := func() bool {
+			_, err := http.Get("ifconfig.co")
+			return err == nil
+		}
+		netConnection := checkConnection()
+		//retrieve offline option from request and cast it into bool
+		offline, _ := req.Options["offline"].(bool)
+
+		if offline || !node.IsOnline || !netConnection {
+			fmt.Println("checking connection")
+			allKeys, err := node.Blockstore.AllKeysChan(ctx)
+			if err != nil {
+				return err
+			}
+			for k := range allKeys {
+				if k == rp.Cid() {
```

Comment:
This is an equivalent of doing `ipfs refs local | grep CID`, which is bit rough :)

There are better ways of checking if a block is in the datastore, for example you could use `ipfs dag stat --offline`:

```
$ ipfs dag stat /ipfs/QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMnR --offline
Size: 119776, NumBlocks: 1

$ ipfs dag stat /ipfs/QmbWqxBEKC3P8tqsKc98xmWNzrzDtRLMiMPL8wBuTGsMns --offline
Error: merkledag: not found
```

Mind replacing this for loop with something like:

```
	_, err := api.Dag().Get(req.Context, rp.Cid())
	if err != nil {
        # getting remote block in offline mode will produce Error: merkledag: not found
		return err
	}
```

Reply:
Thank you for the help @lidel , I will make the changes and re-submitt the pr.